### PR TITLE
Check for corrupted stamp on session::read_json.

### PR DESCRIPTION
**Pull request by karitra**
subj.

**TODO**:
 - [x] tests.

### CODE REVIEW

File: example/eblob_backend.cpp [hunk 1]
```diff
@@ -487,6 +487,23 @@ static int blob_read_new_impl(eblob_backend_config *c,
 		wc.size -= sizeof(ehdr) + ehdr.size;
 		wc.data_offset += sizeof(ehdr) + ehdr.size;
 		record_offset += sizeof(ehdr) + ehdr.size;
+
+		auto data_size = wc.size;
+		auto data_offset = wc.data_offset;
+
+		if (request.read_flags & DNET_READ_FLAGS_DATA) {
```
Comment (shaitan) [inline example/eblob_backend.cpp]: Why is it done inside `if (wc.flags & BLOB_DISK_CTL_EXTHDR) {}` scope?

File: example/eblob_backend.cpp [hunk 2]
```diff
@@ -487,6 +487,23 @@ static int blob_read_new_impl(eblob_backend_config *c,
...
+			// TODO: should be checked `jhdr.capacity > data_size` case?
+			data_size -= jhdr.capacity;
```
Comment (shaitan) [inline example/eblob_backend.cpp]: Why are `data_size` and `data_offset` shifted when `if (request.read_flags & DNET_READ_FLAGS_DATA) {}`?

Comment (karitra) [reply example/eblob_backend.cpp]: WIP artifact, fixed.

File: tests/corrupted_stamp_test.cpp [hunk 3]
```diff
@@ -418,16 +418,17 @@ void test_read_corrupted_stamp(ioremap::elliptics::newapi::session &s, const dne
...
-		BOOST_REQUIRE_EQUAL(r.status(), -EILSEQ);
-	}
-
-	for (const auto &r: s.bulk_read_json(all_illigal_stamp)) {
-		BOOST_REQUIRE_EQUAL(r.status(), 0);
-	}
-
-	for (const auto &r: s.bulk_read_data(all_illigal_stamp)) {
-		BOOST_REQUIRE_EQUAL(r.status(), -EILSEQ);
+	if (all_illigal_stamp.empty() == false) {
```
Comment (shaitan) [inline tests/corrupted_stamp_test.cpp]:
```suggestion
	if (!all_illigal_stamp.empty()) {
```

File: tests/corrupted_stamp_test.cpp [hunk 4]
```diff
@@ -211,14 +212,16 @@ void test_read_corrupted_stamp(ioremap::elliptics::newapi::session &s, const dne
 		newapi::async_write_result async_result;
 	};
 
-	const auto make_write_result = [&] (const std::string &key_prefix,
+	const auto make_write_result = [&] (const int id,
```
Comment (shaitan) [inline tests/corrupted_stamp_test.cpp]: Why has `id` different type? `unsigned` vs `const int`

**Pull request merged**